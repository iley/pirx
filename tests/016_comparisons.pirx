func main(): int {
  var a: int;
  var b: int;
  
  a = 5;
  b = 3;
  
  // Test equality operators
  printf("5 == 3: %d\n", a == b);
  printf("5 != 3: %d\n", a != b);
  printf("5 == 5: %d\n", a == 5);
  
  // Test comparison operators
  printf("5 < 3: %d\n", a < b);
  printf("5 > 3: %d\n", a > b);
  printf("5 <= 3: %d\n", a <= b);
  printf("5 >= 3: %d\n", a >= b);
  printf("5 <= 5: %d\n", a <= 5);
  printf("5 >= 5: %d\n", a >= 5);
  
  return 0;
} 
