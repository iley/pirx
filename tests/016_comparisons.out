5 == 3: 0
5 != 3: 1
5 == 5: 1
5 < 3: 0
5 > 3: 1
5 <= 3: 0
5 >= 3: 1
5 <= 5: 1
5 >= 5: 1
